R/process_lookup_homelessness.R

```diff
@@ -0,0 +1,114 @@
+# It would probably be simplest to create a lookup from the Homelessness data of one row per CHI +
+#   start dates of all applications. This can then be matched to episode and individual to easily create the hl1_in_fy flag(s)
+# (if someone is in the lookup -> 1, otherwise -> 0). For the episode file, it is slightly more complicated to create the other flags,
+# either matching on all the data by CHI and then doing some comparison of dates (as in SPSS),
+# or using some of the fancy new joins (dplyr 1.1.0 joins) and doing it that way.
+#
+#
+
+
+# data <- slfhelper::read_slf_episode("1718", c("anon_chi", "record_keydate1", "record_keydate2", "recid"))
+
+# year <- "1718"
+
+
+#' create homelessness lookup
+#' @description reads in homelessness extract and selects CHIs and flags them as homeless in FY
+#'
+#' @param year The year to process, in FY format.
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @export
+#' @family process extracts
+#'
+create_homelessness_lookup <- function(year) {
+  homelessness_lookup <- read_file(get_source_extract_path(year, "Homelessness")) %>%
+    dplyr::distinct(chi, record_keydate1, record_keydate2) %>%
+    tidyr::drop_na(chi) %>%
```

Comment:
You will need to specify variables something like this, to avoid a build note. You can run a CHECK (`devtools::check()`) to check for these things, or use [`lintr`](https://lintr.r-lib.org/) (e.g. lintr::lint("R/process_lookup_homelessness.R") which will be quicker but will highlight more 'issues' which don't necessarily need fixing/changing but are generally good-practice.
```suggestion
    dplyr::distinct(.data$chi, .data$record_keydate1, .data$record_keydate2) %>%
    tidyr::drop_na(.data$chi) %>%
```